h5t_types.go

```diff
@@ -438,14 +444,18 @@ func NewDataTypeFromType(t reflect.Type) (*Datatype, error) {
 		if err != nil {
 			return nil, err
 		}
+		var ptrPathLen int
 		n := t.NumField()
 		for i := 0; i < n; i++ {
 			f := t.Field(i)
-			var field_dt *Datatype = nil
+			var field_dt *Datatype
```

Comment:
I know you didn't wrote this in the first place, but maybe change to `fieldDt`? Maybe I should also open an issue for moving to Go-style Camel Case?

Reply:
There are a collection of things like this that I'd like to fix up. Please do file that issue.